### PR TITLE
Exegol 4.0.5

### PR DESCRIPTION
**Pull request by Dramelac**
# New feature / options

- New shell logging option: On container creation add `--log` or `-l` on log every session on `/workspace/logs/`
- New file sharing 'tricks' between host and container. (Use `--update-fs` or `-fs` to update permision `g+s` on existing directory and sub-directory) #103 
- New sponsor
- New label system allowing the wrapper to integrate additional features

# Bug fix

- Conflict between nightly images
- Fix default image selection (not printed in the table)
- Filter non updatable images
- #105 
- Upgrade docker SDK, now support python 3.10: #108
- #112

### CODE REVIEW

File: exegol/model/ContainerConfig.py [hunk 1]
```diff
@@ -354,6 +382,20 @@ def disableExegolResources(self):
...
 
+    def enableShellLogging(self):
+        """Procedure to enable exegol shell logging feature"""
+        if not self.__shell_logging:
+            logger.verbose("Config: Enabling shell logging")
+            self.__shell_logging = True
+            self.addLabel(self.__label_features.get('enableShellLogging'), "Enabled")
+
+    def __disableShellLogging(self):
+        """Procedure to enable exegol shell logging feature"""
```
Comment (ShutdownRepo) [inline exegol/model/ContainerConfig.py:393]: Small typo here, enable -> disable
Also, it seems the enable and disable functions have different naming format, was it on purpose?

File: exegol/model/ExegolContainer.py [hunk 2]
```diff
@@ -25,10 +25,20 @@ def __init__(self, docker_container: Container, model: Optional[ExegolContainerT
...
+            image_name = ""
+            try:
+                # Try to find the attached docker image
+                docker_image = docker_container.image
+            except ImageNotFound:
+                # If it is not found, the user has probably forcibly deleted it manually
+                logger.warning(f"Some images were forcibly removed by docker when they were used by existing containers!")
+                logger.error(f"The '{docker_container.name}' containers might not work properly anymore and should also be deleted and recreated with a new image.")
+                docker_image = None
+                image_name = "[red bold]BROKEN[/red bold]"
```
Comment (ShutdownRepo) [inline exegol/model/ExegolContainer.py:37]: could this be changed to "missing image", for clarity's sake?

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]

File: exegol/model/ExegolContainer.py (hunk 2 above)
Comment (Dramelac) [reply exegol/model/ExegolContainer.py:37]: For UI size reasons, we decided not to change.

File: exegol/model/ContainerConfig.py (hunk 1 above)
Comment (ShutdownRepo) [reply exegol/model/ContainerConfig.py]: got it :+1:

Comment (Dramelac) [reply exegol/model/ContainerConfig.py]: Type fixed in #[81963dc](https://github.com/ShutdownRepo/Exegol/pull/115/commits/81963dc8f1acabbb8b749e680c505fa045d8a8b1)

File: exegol/model/ExegolContainer.py (hunk 2 above)
Comment (ShutdownRepo) [reply exegol/model/ExegolContainer.py:37]: [internal discussion] keeping for better looking reasons, "missing image" would be too long

**Review by ShutdownRepo — APPROVED** [review]

**Pull request merged**